dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -105,10 +191,28 @@ public UfsUrl getChildURL(String childName) {
   }
 
   public String getFullPath() {
-    return Strings.join(mProto.getPathComponentsList(), AlluxioURI.SEPARATOR.charAt(0));
+    // TODO(Tony Sun): Is it correct to return a '/' with empty pathComponents?
+    return "/" + Strings.join(mProto.getPathComponentsList(), AlluxioURI.SEPARATOR.charAt(0));
   }
 
   public AlluxioURI toAlluxioURI() {
-    return new AlluxioURI(mProto.getScheme(), Authority.fromString(mProto.getAuthority()), getFullPath());
+    return new AlluxioURI(mProto.getScheme(),
+        Authority.fromString(mProto.getAuthority()), getFullPath());
+  }
+
+  public int getDepth() {
+    return getPathComponents().size();
+  }
+
+  public String getName() {
+    List<String> pathComponents = getPathComponents();
+    return pathComponents.get(pathComponents.size() - 1);
+  }
+
+  //TODO(Tony Sun): Add isAncestorOf() method.
+
+  public boolean isRoot() {
+    // TODO(Tony Sun): In AlluxioURI here remains judging hasAuthority(), why?
+    return getFullPath().equals(PATH_SEPERATOR) || getFullPath().isEmpty();
```

Comment:
something like this?
```suggestion
    return mProto.getPathComponentsSize();
```